### DISCUSSION TITLE
How to upgrade code-server in synology docker?

### DISCUSSION THREAD

**Discussion by patialo** [coder/code-server#5124] [Q&A] [answered]
My code-server was installed in  synology docker , but not using command line.
How can i upgrade code-server  to ensure that my data is not lost？ 
My idea is to delete the container, upgrade the image, and then reinstall it. I don't know if it will lead to data loss，when use it in ubuntu，`/.local/share/....` is independent of the code-server‘s system，but docker container seems not.

**Comment by jsjoeio** [comment]
Hmm...I am not sure. I assume any data in the container would be lost but I'm not familiar with synology so I can't say forsure.

**Comment by lg-nineoneninethreenine** [comment, marked as answer]
Well, I think if you backup the files locally and proceed by downloading and booting the latest image, it might work. I think you can use rclone, get everything there on GitHub, reinstall the image and move your files back as well as set up your files on Synology.

**Reply by patialo** [reply]
This is a workaround, but I hope it works once and for all.
Maybe map /home/coder/.local/share out of docker? Or map the entire /home folder like mapping /home/coder/project
Please tell me if this is possible @jsjoeio

**Reply by patialo** [reply]
It’s working! Map /.local/share  like /project make extensions and settings  saved in my synology file station successfully
`-v docker/code-server/share /home/coder/.local/share`
